app/src/main/report/components/OneTimeLinkContext.tsx

```diff
@@ -57,4 +58,14 @@ export function OneTimeLinkContext(WrappedComponent: ComponentConstructor<OneTim
                 children={this.props.children}/>;
         }
     })
+}
+
+function appendAccessToken(url: string, token: string) {
+    url = buildReportingURL(url);
+    if (url.indexOf("?") > 0) {
+        return url + "&access_token=" + token;
+    }
+    else{
+        return url + "?access_token=" + token;
+    }
```

Comment:
Again, is there a library in JS to handle this nicely? Just wondering about edge cases, like more than `?` in a URL or something weird like that.